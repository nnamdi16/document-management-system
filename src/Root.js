/* eslint-disable no-undef */
/* eslint-disable no-shadow */
import React from 'react';
import { Provider } from 'react-redux';
import { createStore, applyMiddleware } from 'redux';
import reducers from 'reducers';
import reduxThunk from 'redux-thunk';

const initialState = {
	auth: { authenticated: localStorage.getItem('token') }
};
export default ({ children, state = initialState }) => {
	const store = createStore(reducers, state, applyMiddleware(reduxThunk));
	return <Provider store={ store }>{children}</Provider>;
};
