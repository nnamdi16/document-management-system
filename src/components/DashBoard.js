import React, { Component } from 'react';
import requireAuth from 'components/requireAuth';

class DashBoard extends Component {
	render() {
		return (
    <div>
        <h1>Welcome to your DMS dashboard</h1>
    </div>
		);
	}
}

export default requireAuth(DashBoard);
