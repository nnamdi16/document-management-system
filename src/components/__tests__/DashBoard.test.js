/* eslint-disable no-undef */
import React from 'react';
// import { mount } from 'enzyme';
import 'jest-enzyme';
import '../../../setupTests';
import DashBoard from 'components/DashBoard';
import { mount } from 'enzyme';
import Root from '../../Root';

describe('Shallow render the Dashboard', () => {
	it('shallow render the Dashboard', () => {
		const wrapper = mount(
    <Root>
        <DashBoard />
    </Root>
		);
		const welcome = <div>Welcome to your DMS dashboard</div>;
		expect(wrapper.contains(welcome)).toEqual(true);
	});
});
