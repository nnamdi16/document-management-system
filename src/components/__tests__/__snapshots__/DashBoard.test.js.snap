// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shallow render the Dashboard shallow render the Dashboard 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ConnectFunction />
</Provider>
`;

exports[`Shallow render the Dashboard shallow render the Dashboard 2`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <ConnectFunction />
</Provider>
`;
